std/file.d

```diff
@@ -539,10 +539,20 @@ version(Posix) private void writeImpl(const(char)[] name, const(FSChar)* namez,
     cenforce(fd != -1, name, namez);
     {
         scope(failure) core.sys.posix.unistd.close(fd);
+
         immutable size = buffer.length;
-        cenforce(
-            core.sys.posix.unistd.write(fd, buffer.ptr, size) == size,
-            name, namez);
+        immutable chks = size_t.max / 2 + 1;
```

Comment:
Where does this number come from ? [This post](http://programmers.stackexchange.com/a/171326) suggest 8192 for nix and 65536 on Win.
HDD are buffered anyway but using a too big number may cause page allocations, which would kill the gain obtained by decreasing the system calls. [see also this](http://stackoverflow.com/a/32172941).

So I don't know, maybe 8192 * 2 might be a good compromise.